ecl/resource_ecl_provider_connectivity_tenant_connection_request_v2.go

```diff
@@ -0,0 +1,274 @@
+	d.Set("tenant_id", request.TenantID)
+	d.Set("name_other", request.NameOther)
+	d.Set("description_other", request.DescriptionOther)
+	d.Set("tags_other", request.TagsOther)
+	d.Set("tenant_id_other", request.TenantIDOther)
+	d.Set("network_id", request.NetworkID)
+	d.Set("approval_request_id", request.ApprovalRequestID)
+
+	return nil
+}
+
+func resourceProviderConnectivityTenantConnectionRequestV2Update(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	var hasChange bool
+	var updateOpts tenant_connection_requests.UpdateOpts
+
+	if d.HasChange("name") {
+		hasChange = true
+		updateOpts.Name = d.Get("name").(string)
+	}
+
+	if d.HasChange("description") {
+		hasChange = true
+		updateOpts.Description = d.Get("description").(string)
+	}
+
+	if d.HasChange("tags") {
+		hasChange = true
+		updateOpts.Tags = getTags(d, "tags")
+	}
+
+	if d.HasChange("name_other") {
+		hasChange = true
+		updateOpts.NameOther = d.Get("name_other").(string)
+	}
+
+	if d.HasChange("description_other") {
+		hasChange = true
+		updateOpts.DescriptionOther = d.Get("description_other").(string)
+	}
+
+	if d.HasChange("tags_other") {
+		hasChange = true
+		updateOpts.TagsOther = getTags(d, "tags_other")
+	}
+
+	if hasChange {
+		r := tenant_connection_requests.Update(client, d.Id(), updateOpts)
+		if r.Err != nil {
+			return fmt.Errorf("error updating ECL Provider Connectivity Tenant Connection Request: %s", r.Err)
+		}
+		log.Printf("[DEBUG] Tenant Connection Request has successfully updated.")
+	}
+
+	return resourceProviderConnectivityTenantConnectionRequestV2Read(d, meta)
+}
+
+func resourceProviderConnectivityTenantConnectionRequestV2Delete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	if err := tenant_connection_requests.Delete(client, d.Id()).ExtractErr(); err != nil {
+		return fmt.Errorf("error deleting ECL Provider Connectivity Tenant Connection Request: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Target:     []string{"DELETED"},
+		Refresh:    waitForTenantConnectionRequestStateDelete(client, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error deleting ECL tenant connection request (%s): %s", d.Id(), err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func getTags(d *schema.ResourceData, tagName string) map[string]string {
+	rawTags := d.Get(tagName).(map[string]interface{})
+	tags := map[string]string{}
+	for key, value := range rawTags {
+		if v, ok := value.(string); ok {
```

Comment:
You should not omit values that type assertion fails. You must return validation error if an user set non-string value.

Reply:
I'll do it.